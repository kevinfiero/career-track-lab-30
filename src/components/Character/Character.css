img{
  width: 6rem;
  height: 7rem;
}

figcaption{
  width: 8rem;
  align-content: center;
}

figure{
  text-align: center;
  display: flex;
  flex-direction: column;
}
