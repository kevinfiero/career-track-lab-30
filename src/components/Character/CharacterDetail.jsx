import React from 'react';
import PropTypes from 'prop-types';
import './CharacterDetail.css';

const Character = (
  { 
    image, 
    name,
    quote,
    gender,
    personality,
    species,
    birthday,
    phrase,
    skill
  }
) => (
  <section>
    <figure>
      <img className="detailImage" src={image} />
      <figcaption>
        <div>
          <label id="name-label">Name:</label>
          <h1 aria-labelledby="name-label">{name}</h1>
        </div>
        <div>
          <label>Quote:</label>
          <h1>{quote}</h1>
        </div>
        <div>
          <label>Gender:</label>
          <h1>{gender}</h1>
        </div>
        <div>
          <label>Personality:</label>
          <h1>{personality}</h1>
        </div>
        <div>
          <label>Species:</label>
          <h1>{species}</h1>
        </div>
        <div>
          <label>Birthday:</label>
          <h1>{birthday}</h1>
        </div>
        <div>
          <label>Phrase:</label>
          <h1>{phrase}</h1>
        </div>
        <div>
          <label>Skill:</label>
          <h1>{skill}</h1>
        </div>
      </figcaption>
    </figure>
  </section>
);

Character.propTypes = {
  image: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  quote: PropTypes.string.isRequired,
  gender: PropTypes.string.isRequired,
  personality: PropTypes.string.isRequired,
  species: PropTypes.string.isRequired,
  birthday: PropTypes.string.isRequired,
  phrase: PropTypes.string.isRequired,
  skill: PropTypes.string.isRequired
};

export default Character;
