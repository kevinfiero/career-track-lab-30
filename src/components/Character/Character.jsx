import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import './Character.css';

const Character = ({ image, name, id }) => (
  <Link to={`./${id}`}>
    <figure>
      <img src={image} />
      <figcaption>
        <h1>{name}</h1>
      </figcaption>
    </figure>
  </Link>
);

Character.propTypes = {
  image: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired
};

export default Character;
