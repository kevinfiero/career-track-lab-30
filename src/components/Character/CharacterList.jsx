import React from 'react';
import PropTypes from 'prop-types';
import Character from './Character';
import './CharacterList.css';

const CharacterList = ({ characters }) => {
  const characterElements = characters.map((character) => (
    <li key={character.id}>
      <Character 
        image={character.image}
        name={character.name}
      />
    </li>
  ));

  return <ul>{characterElements}</ul>;

};

CharacterList.propTypes = {
  characters: PropTypes.array.isRequired
};

export default CharacterList;
