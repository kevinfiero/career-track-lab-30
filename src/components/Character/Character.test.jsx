import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { render, cleanup } from '@testing-library/react';
import Character from './Character';

describe('Character component', () => {
  afterEach(() => cleanup());
  it('renders Character', () => {
    const { asFragment } = render(
      <Router>
        <Character 
          id="5f5fb4bbbfd05c2aed82e460"
          name = "Admiral" 
          image="https://vignette.wikia.nocookie.net/animalcrossing/images/e/ed/Admiral_NH.png/revision/latest?cb=20200802081138"
        />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
