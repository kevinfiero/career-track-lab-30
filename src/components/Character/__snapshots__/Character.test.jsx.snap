// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders Character 1`] = `
<DocumentFragment>
  <a
    href="/undefined"
  >
    <figure>
      <img
        src="https://vignette.wikia.nocookie.net/animalcrossing/
 images/e/ed/Admiral_NH.png/revision/latest?cb=20200802081138"
      />
      <figcaption>
        <h1>
          Admiral
        </h1>
      </figcaption>
    </figure>
  </a>
</DocumentFragment>
`;
